src/app/dataset-block/metadata-block/components/block-navigation/block-navigation.component.html

```diff
@@ -1 +1,72 @@
-<p>block-navigation works!</p>
+<div class="text-center mb-4 position-relative align-items-center">
+    <input
+        class="form-control placeholder-padding-left"
+        type="text"
+        name="search-hash"
+        [(ngModel)]="searchHash"
+        autocomplete="off"
+        placeholder="Search by hash"
+    />
+    <svg-icon name="search"></svg-icon>
+</div>
+<div class="mb-4">
+    <div class="select-block w-100">
+        <div
+            class="form-control input-group justify-content-left align-items-center"
+        >
+            <p class="m-auto position-absolute">Filter:</p>
+            <select
+                [(ngModel)]="eventType"
+                class="border-0 w-100 py-1 outline-none"
+                data-test-id="event-type-sort-option"
+                id="sortOptionsEventType"
+            >
+                <option
+                    class="py-1"
+                    *ngFor="let option of sortOptions"
+                    value="{{ option }}"
+                >
+                    {{ option }}
+                </option>
+            </select>
+        </div>
+    </div>
+</div>
+<!-- <pre>{{ datasetHistory | json }}</pre> -->
+<div class="d-flex flex-column gap-4" *ngIf="datasetHistory?.history">
```

Comment:
When `datasetHistory` can be null ? If it can, it should be reflected by it's type.

Reply:
Fixed